Pyrado/pyrado/algorithms/stopping_criteria/predefined_criteria.py

```diff
@@ -0,0 +1,117 @@
+
+from pyrado.algorithms.stopping_criteria.stopping_criterion import StoppingCriterion
+
+
+class AlwaysStopStoppingCriterion(StoppingCriterion):
+    """Stopping criterion that is always met."""
+
+    def __repr__(self) -> str:
+        return "AlwaysStopStoppingCriterion"
+
+    def __str__(self) -> str:
+        return "True"
+
+    def is_met(self, algo) -> bool:
+        """Returns `True`."""
+        return True
+
+
+class NeverStopStoppingCriterion(StoppingCriterion):
+    """Stopping criterion that is never met."""
+
+    def __repr__(self) -> str:
+        return "NeverStopStoppingCriterion"
+
+    def __str__(self) -> str:
+        return "False"
+
+    def is_met(self, algo) -> bool:
+        """Returns `False`."""
+        return False
+
+
+class CustomStoppingCriterion(StoppingCriterion):
+    """Custom stopping criterion that takes an arbitrary callable to evaluate."""
+
+    def __init__(self, criterion_fn: Callable[[Any], bool], name: Optional[str] = None):
+        """
+        Constructor.
+
+        :param criterion_fn: signature `[Algorithm] -> bool`; gets evaluated when `is_met` is called; allows for custom
+                             functionality, e.g. if an algorithm requires special treatment; the given algorithm is the
+                             same that was passed to the `is_met` method
+        :param name: name of the stopping criterion, used for `str(..)` and ´repr(..)`
+        """
+        self._criterion_fn = criterion_fn
+        self._name = name
+
+    def __repr__(self) -> str:
+        return f"CustomStoppingCriterion[_criterion_fn={repr(self._criterion_fn)}; name={self._name}]"
+
+    def __str__(self) -> str:
+        return "Custom" if self._name is None else self._name
+
+    def is_met(self, algo) -> bool:
+        """Invokes the criterion function that was passed to the constructor."""
+        return self._criterion_fn(algo)
+
+
+class IterCountStoppingCriterion(StoppingCriterion):
+    """Uses the iteration number as a stopping criterion, i.e. sets a maximum number of iterations."""
+
+    def __init__(self, max_iter: int):
+        """
+        Constructor.
+
+        :param max_iter: maximum number of iterations
+        """
+        self._max_iter = max_iter
+
+    def is_met(self, algo) -> bool:
+        """Returns whether the current iteration number os greater than or equal to the maximum number of iterations."""
+        return algo.curr_iter >= self._max_iter
+
+
+class SampleCountStoppingCriterion(StoppingCriterion):
+    """Uses the sampler count as a stopping criterion, i.e. sets a maximum number samples."""
+
+    def __init__(self, max_sample_count: int):
+        """
+        Constructor.
+
+        :param max_sample_count: maximum sample count
+        """
+        self._max_sample_count = max_sample_count
+
+    def is_met(self, algo) -> bool:
+        """Returns whether the current sample count is greater than or equal to the maximum sample count."""
```

Comment:
You can (and please do it) delete doc strings on functions inherited from a superclass *if* they have the same signature *and* they are not super special.